src/and.js

```diff
@@ -19,5 +23,5 @@ var _curry2 = require('./internal/_curry2');
  *      R.and(null, ''); => null
  */
 module.exports = _curry2(function and(a, b) {
-  return a && b;
+  return _hasMethod('and', a) ? a.and(b) : a && b;
```

Comment:
`&&` is not commutative for types with multiple "true" values:

```javascript
1 && 2;  // => 2
2 && 1;  // => 1
```

It's important that `R.and` behave similarly:

```javascript
Just(1).and(Just(2));  // => Just(2)
Just(2).and(Just(1));  // => Just(1)
```